src/topology/bounded_continuous_function.lean

```diff
@@ -22,32 +22,51 @@ universes u v w
 variables {α : Type u} {β : Type v} {γ : Type w}
 
 /-- The type of bounded continuous functions from a topological space to a metric space -/
-def bounded_continuous_function (α : Type u) (β : Type v) [topological_space α] [metric_space β] :
+structure bounded_continuous_function
+  (α : Type u) (β : Type v) [topological_space α] [metric_space β] extends continuous_map α β :
   Type (max u v) :=
-{f : α → β // continuous f ∧ ∃C, ∀x y:α, dist (f x) (f y) ≤ C}
+(bounded' : ∃C, ∀x y:α, dist (to_fun x) (to_fun y) ≤ C)
 
-local infixr ` →ᵇ `:25 := bounded_continuous_function
+localized "infixr ` →ᵇ `:25 := bounded_continuous_function" in bounded_continuous_function
 
 namespace bounded_continuous_function
 section basics
 variables [topological_space α] [metric_space β] [metric_space γ]
 variables {f g : α →ᵇ β} {x : α} {C : ℝ}
 
-instance : has_coe_to_fun (α →ᵇ β) :=  ⟨_, subtype.val⟩
+instance : has_coe_to_fun (α →ᵇ β) :=  ⟨_, λ f, f.to_fun⟩
+
+protected lemma bounded (f : α →ᵇ β) : ∃C, ∀x y:α, dist (f x) (f y) ≤ C := f.bounded'
+protected lemma continuous (f : α →ᵇ β) : continuous f := f.to_continuous_map.continuous
 
 lemma bounded_range : bounded (range f) :=
-bounded_range_iff.2 f.2.2
+bounded_range_iff.2 f.bounded
 
-/-- If a function is continuous on a compact space, it is automatically bounded,
-and therefore gives rise to an element of the type of bounded continuous functions -/
-def mk_of_compact [compact_space α] (f : α → β) (hf : continuous f) : α →ᵇ β :=
-⟨f, hf, bounded_range_iff.1 $ bounded_of_compact $ compact_range hf⟩
+/-- A continuous function with an explicit bound is a bounded continuous function. -/
+def mk_of_bound (f : C(α, β)) (C : ℝ) (h : ∀ x y:α, dist (f x) (f y) ≤ C) : α →ᵇ β :=
+⟨f, ⟨C, h⟩⟩
```

Comment:
This could be namespaced `continuous_map` (at least as an alias) to allow for dot notation, same for `mk_of_compact`.  Not sure about names, maybe `continuous_map.mk_bounded` and `continuous_map.mk_bounded_of_compact`.

Reply:
Also, `C` can be implicit here I think.

Reply:
I'd prefer to leave it explicit. I can image knowing the bound, and writing `mk_of_bound 37`, and expecting to be left with the proof obligation.